src/utils/PaymentsConverter.ts

```diff
@@ -0,0 +1,236 @@
+/**
+ * Payments Converter
+ * Data Converter for Payments Request\Responses between PagoPa and BackendAPP types
+ */
+import { Either, left, right } from "fp-ts/lib/Either";
+import {
+  codificaInfrastrutturaPSPEnum,
+  IcdInfoWispInput,
+  InodoAttivaRPTInput,
+  InodoAttivaRPTOutput,
+  InodoVerificaRPTInput,
+  InodoVerificaRPTOutput,
+  PPTPortTypes
+} from "italia-pagopa-api/dist/wsdl-lib/PagamentiTelematiciPspNodoservice/PPTPort";
+import * as uuid from "uuid";
+import { PagoPaConfig } from "../Configuration";
+import { ControllerError } from "../enums/ControllerError";
+import { PaymentsActivationRequest } from "../types/controllers/PaymentsActivationRequest";
+import { PaymentsActivationResponse } from "../types/controllers/PaymentsActivationResponse";
+import { PaymentsCheckRequest } from "../types/controllers/PaymentsCheckRequest";
+import { PaymentsCheckResponse } from "../types/controllers/PaymentsCheckResponse";
+import { PaymentsStatusUpdateRequest } from "../types/controllers/PaymentsStatusUpdateRequest";
+import { CodiceContestoPagamento } from "../types/PagoPaTypes";
+
+// Convert PaymentsCheckRequest (controller) to PaymentsCheckRequestPagoPa (PagoPa API)
+export function getPaymentsCheckRequestPagoPa(
+  pagoPaConfig: PagoPaConfig,
+  paymentsCheckRequest: PaymentsCheckRequest,
+  codiceContestoPagamento: CodiceContestoPagamento
+): Either<Error, InodoVerificaRPTInput> {
+  return right({
+    identificativoPSP: pagoPaConfig.IDENTIFIER.IDENTIFICATIVO_PSP,
+    identificativoIntermediarioPSP:
+      pagoPaConfig.IDENTIFIER.IDENTIFICATIVO_INTERMEDIARIO_PSP,
+    identificativoCanale: pagoPaConfig.IDENTIFIER.IDENTIFICATIVO_CANALE,
+    password: pagoPaConfig.IDENTIFIER.TOKEN,
+    codiceContestoPagamento,
+    codificaInfrastrutturaPSP: codificaInfrastrutturaPSPEnum.QR_CODE,
+    codiceIdRPT: {
+      CF: paymentsCheckRequest.codiceIdRPT.CF,
+      AuxDigit: paymentsCheckRequest.codiceIdRPT.AuxDigit,
+      CodStazPA: paymentsCheckRequest.codiceIdRPT.CodStazPA,
+      CodIUV: paymentsCheckRequest.codiceIdRPT.CodIUV
+    }
+  });
+}
+
+// Convert PaymentsCheckResponsePagoPa (PagoPa API) to PaymentsCheckResponse (controller)
+export function getPaymentsCheckResponse(
+  iNodoVerificaRPTOutput: InodoVerificaRPTOutput,
+  codiceContestoPagamento: CodiceContestoPagamento
+): Either<Error, PaymentsCheckResponse> {
+  if (
+    iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.esito ===
+    PPTPortTypes.Esito.KO
+  ) {
+    return left(new Error(ControllerError.REQUEST_REJECTED));
+  }
+  const errorOrPaymentCheckResponse = PaymentsCheckResponse.decode({
+    importoSingoloVersamento:
+      iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
```

Comment:
please use an alias for iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA

`const datiPagamentoPA = iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA`

Reply:
why? It's a direct mapping, used just there

Reply:
1. to save keystrokes and help code review
2. it's more efficient in terms of performance (but this is negligible)

moreover consider using object composition with spread operators to do this kind of assignments between objects that have lot of properties in common. it looks like this can be reduced to:

```ts
const datiPagamento = iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA;
const versamento = { ...datiPagamento, enteBeneficiario: {
    ...datiPagamento.enteBeneficiario,
    codiceIdentificativoUnivoco: datiPagamento.enteBeneficiario
  }
}
```
another option is to use [lens](https://github.com/gcanti/monocle-ts#frompath)